pytest-server-fixtures/pytest_server_fixtures/base.py

```diff
@@ -13,23 +13,65 @@
 import traceback
 from datetime import datetime
 import logging
+import random
 
 from six import string_types
 
 from pytest_server_fixtures import CONFIG
 from pytest_shutil.workspace import Workspace
 
 log = logging.getLogger(__name__)
+_SESSION_HOST = None
 
 
-def get_ephemeral_port():
-    """ Get an ephemeral socket at random from the kernel
+def get_ephemeral_host():
     """
-    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    s.bind(('127.0.0.1', 0))
-    port = s.getsockname()[1]
-    s.close()
-    return port
+    Returns a random IP in the 127.0.0.0/8 whcih we will
+    use as the basis for ports in this test-run
+    """
+    global _SESSION_HOST
+    if _SESSION_HOST:
+        return _SESSION_HOST[0]
+
+    # return a host in the 127.x.x.x range
+    while True:
+        try:
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            host = '127.{}.{}.{}'.format(random.randrange(1, 255),
+                                         random.randrange(1, 255),
+                                         random.randrange(2, 255),)
+            s.bind((host, 5000))
+            s.listen(0)
+            _SESSION_HOST = (host, s)
+            return _SESSION_HOST[0]
+        except socket.error:
+            pass
+
+def get_ephemeral_port(host=get_ephemeral_host()):
```

Comment:
Please use:

    def get_ephemeral_port(host=None):
        if host is None:
            host = get_ephemeral_host()

Reason being, default arguments are evaluated on import.